packages/api/src/lib/university-admin-dashboard.ts

```diff
@@ -0,0 +1,128 @@
+import axios from 'axios';
+import { getAuthHeaders } from './lmp';
+export interface SemesterData {
+  universityId?: string;
+  instanceId?: string;
+  semesterStart: string;
+  semesterEnd: string;
+  lectureStartDate: string;
+  lectureEndDate: string;
+  timeZone: string;
+}
+
+export interface Holiday {
+  date: string;
+  name: string;
+}
+
+export interface UploadHolidaysRequest {
+  universityId: string;
+  instanceId: string;
+  holidays: Holiday[];
+}
+
+export interface EditHolidayRequest {
+  universityId: string;
+  instanceId: string;
+  originalDate: string;
+  updatedHoliday: Holiday;
+}
+
+export interface DeleteSingleHolidayRequest {
+  universityId: string;
+  instanceId: string;
+  dateToDelete: string;
+}
+
+const SEMESTER_BASE_URL = '/api/university-admin/semester-info';
+
+export async function getSemesterInfo(universityId: string, instanceId: string) {
+  const response = await axios.get(`${SEMESTER_BASE_URL}/get-semester-info`, {
+    params: { universityId, instanceId },
+    headers: getAuthHeaders(),
+  });
+  return response.data.semesterInfo as SemesterData[];
+}
+
+export async function createSemester(data: SemesterData) {
+  const response = await axios.post(`${SEMESTER_BASE_URL}/create-semester`, data, {
+    headers: getAuthHeaders(),
+  });
+  return response.data;

```

Comment:
There is no need to return anything for API requests that dont return any data
```suggestion
  await axios.post(`${SEMESTER_BASE_URL}/create-semester`, data, {
    headers: getAuthHeaders(),
  });
```